src/rules/no-unresolved.js

```diff
@@ -24,6 +24,11 @@ module.exports = {
 
   create(context) {
     function checkSourceValue(source) {
+      // ignore type-only imports
+      if (source.parent.importKind === 'type') {
+        return;
+      }
```

Comment:
can we get some tests for this? ideally both using `import type` and also regular `import`

Reply:
You bet -- I cargo-culted these tests from some of the other type-only tests but they seem to be failing on Node *.[45]? I don't recognize the pattern ... I'll keep investigating. Do you spy anything obvious?
https://github.com/import-js/eslint-plugin-import/blob/35e14312a47812816fdd618c5dbceddc809ceaf2/tests/src/rules/no-unresolved.js#L418-L436

Also, any idea what's with the "TypeScript TypeScript TypeScript" in `1) TypeScript TypeScript TypeScript no-unresolved ignore type-only valid import type { JSONSchema7Type } from "@types/json-schema";:`? https://github.com/import-js/eslint-plugin-import/pull/2220/checks?check_run_id=3576242736#step:5:72

Reply:
I think I solved it: `parser !== require.resolve('typescript-eslint-parser')` like https://github.com/import-js/eslint-plugin-import/blob/7784948fc5024476f0861b7b8bb20603f28cf809/tests/src/rules/no-duplicates.js#L420

Tests are passing now -- did I do it right?